Watchman.Tests/Dynamo/DynamoAlarmTests.cs

```diff
@@ -0,0 +1,199 @@
+                    .ReturnsAsync(new DescribeTableResponse()
+                    {
+                        Table = table
+                    });
+
+            }
+
+            return fakeDynamo.Object;
+        }
+
+        [Test]
+        public async Task IgnoresNamedEntitiesThatDoNotExist()
+        {
+            // arrange
+
+            var stack = new Mock<ICloudformationStackDeployer>();
+
+            var dynamoClient = CreateDynamoClientForTables(new[]
+            {
+                new TableDescription()
+                {
+                    TableName = "first-dynamo-table",
+                    ProvisionedThroughput = new ProvisionedThroughputDescription()
+                    {
+                        ReadCapacityUnits = 10,
+                        WriteCapacityUnits = 10
+                    }
+                }
+            });
+
+            var source = new TableDescriptionSource(dynamoClient);
+
+            var creator = new CloudFormationAlarmCreator(stack.Object);
+
+            var config = ConfigHelper.CreateBasicConfiguration("test", "group-suffix", "DynamoDb", new List<ResourceThresholds>()
+            {
+                new ResourceThresholds()
+                {
+                    Name = "non-existant-table"
+                }
+            });
+
+            var sut = IoCHelper.CreateSystemUnderTest(
+                source, 
+                new DynamoDbDataProvider(), 
+                new DynamoDbDataProvider(),
+                WatchmanServiceConfigurationMapper.MapDynamoDb,
+                creator, 
+                ConfigHelper.ConfigLoaderFor(config)
+                );
+
+            
+
+            // act
+            
+            await sut.LoadAndGenerateAlarms(RunMode.GenerateAlarms);
+            
+            // assert
+
+            stack
+                .Verify(x => x.DeployStack(
+                    It.IsAny<string>(),
+                    It.IsAny<string>(),
+                    It.IsAny<bool>()
+                    ), Times.Never);
+        }
+
+        [Test]
+        public async Task AlarmCreatedWithCorrectProperties()
+        {
+            // arrange
+
+            var stack = new FakeStackDeployer();
+
+            var dynamoClient = CreateDynamoClientForTables(new[]
+            {
+                new TableDescription()
+                {
+                    TableName = "first-dynamo-table",
+                    ProvisionedThroughput = new ProvisionedThroughputDescription()
+                    {
+                        ReadCapacityUnits = 100,
+                        WriteCapacityUnits = 200
+                    }
+                }
+            });
+
+            var source = new TableDescriptionSource(dynamoClient);
+            var creator = new CloudFormationAlarmCreator(stack);
+
+            var config = ConfigHelper.CreateBasicConfiguration("test", "group-suffix", "DynamoDb", new List<ResourceThresholds>()
```

Comment:
Considering whether it would be be better to actually load a json file here, not sure -so that the the test covers even more of the application. But that change could be done later if we want to.